example/filter.go

```diff
@@ -0,0 +1,205 @@
+/*
```

Comment:
wow, we should not move these files into example.
example is just for testing, people could see how it works.

Reply:
I get it.

Reply:
Done.